### PR TITLE
Add test cases to verify hashing for IPinIP packet

### LINKED ISSUE #7569: [test gap] Add a test case to verify inner IP header of IPinIP packet is used as hash key
**Description**

The test case is to verify the ECMP behavior for IPinIP packet. 
It's expected that the `src_ip`, `dst_ip`, `src_port` and `dst_port` of inner packet are used as hash key.
The test case will run on T1 topology.

**Steps to reproduce the issue:**
1. Generate IPinIP packets, where the src_ip and dst_ip of the outer layer are fixed, and src_ip, dst_ip, src_port and dst_port of inner packet are randomized.
2. Ingress the packet from T0 facing port
3. Count the packets received on each T2 facing port, verify the traffic is balanced among all uplink.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by bingwang-ms**
### Description of PR

Summary:
Fixes #7569
This PR is to add a test case to verity the hashing for IPinIP packet.
It's expected that the `src_ip`, `dst_ip`, `src_port` and `dst_port` of inner packet are used as hash key.
This PR also adds a negative test case to verify the length of inner_frame is not used as hash key.
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?
This PR is to add test cases to verify hashing for IPinIP packet.

#### How did you do it?
Added two test cases
1. test_ipinip_hash to verify the `src_ip`, `dst_ip`, `src_port` and `dst_port` of inner packet are used as hashing key.
2. test_ipinip_hash_negative to verify inner_frame length is not used as hashing key.

#### How did you verify/test it?
The change is verified on a Cisco T1 testbed and Arista T1 testbed.

#### Any platform specific information?
Test case `test_ipinip_hash` is running only on Cisco platform at this moment.
The negative test case `test_ipinip_hash_negative` should run on all platforms.

#### Supported testbed topology if it's a new test case?
Test case `test_ipinip_hash` runs only on T1 topology. 
Test case `test_ipinip_hash_negative` should run on all topologies.

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing ansible/roles/test/files/ptftests/py3/hash_test.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/fib/test_fib.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing ansible/roles/test/files/ptftests/py3/hash_test.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/fib/test_fib.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/fib/test_fib.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
...
 
-from tests.common.fixtures.ptfhost_utils import change_mac_addresses        # noqa F401
-from tests.common.fixtures.ptfhost_utils import remove_ip_addresses         # noqa F401
-from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory     # noqa F401
-from tests.common.fixtures.ptfhost_utils import set_ptf_port_mapping_mode   # noqa F401
-from tests.common.fixtures.ptfhost_utils import ptf_test_port_map_active_active
+from tests.common.fixtures.ptfhost_utils import change_mac_addresses        # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import remove_ip_addresses         # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import copy_ptftests_directory     # lgtm[py/unused-import]
+from tests.common.fixtures.ptfhost_utils import set_ptf_port_mapping_mode   # lgtm[py/unused-import]
```
Comment (wangxin) [inline tests/fib/test_fib.py:11]: `lgtm` is no longer used. The `# noqa F401` is used by pre-commit. Suggest to keep `# noqa F401` and remove `lgtm`.

Comment (bingwang-ms) [reply tests/fib/test_fib.py]: Thanks. Updated

**Review by wangxin — APPROVED** [review]

**Review by lolyu — APPROVED** [review]

**Pull request merged**